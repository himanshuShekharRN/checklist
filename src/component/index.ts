import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
