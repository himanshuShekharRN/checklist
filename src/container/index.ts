import {NationalChecklist} from './nationalChecklist';
import {DepartureDocCard} from './departureDocCard';
import {CustomTab} from './customTab';
import {MyCheckList} from './myChecklist';
import {PreDepartureList} from './preDepartureList';

export {
  NationalChecklist,
  DepartureDocCard,
  CustomTab,
  MyCheckList,
  PreDepartureList,
};
