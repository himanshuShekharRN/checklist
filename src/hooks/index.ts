import {usePreDepartureListData} from './usePreDepartureListData';
import {useChecklistCompletionStatus} from './useChecklistCompletionStatus';

export {useChecklistCompletionStatus, usePreDepartureListData};
