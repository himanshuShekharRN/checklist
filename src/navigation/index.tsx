import {NavigationContainer} from '@react-navigation/native';
import React from 'react';
import {CheckListStackNavigator} from './checkListStackNavigator';

export const Navigation = () => {
  return (
    <NavigationContainer>
      <CheckListStackNavigator />
    </NavigationContainer>
  );
};
