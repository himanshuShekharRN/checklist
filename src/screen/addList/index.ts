import {AddList} from './AddList.screen';

export {AddList};
