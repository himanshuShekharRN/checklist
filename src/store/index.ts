import createSagaMiddleware from 'redux-saga';
import {configureStore} from '@reduxjs/toolkit';
import {persistReducer, persistStore} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

import {appReducer} from './reducer';
import {rootSaga} from '../network/saga';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: ['checkListReducer', 'departureCheckListReducer'],
};

const sagaMiddleware = createSagaMiddleware();
const persistedReducer = persistReducer(persistConfig, appReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: [sagaMiddleware],
});

export const persistor = persistStore(store);
sagaMiddleware.run(rootSaga);
