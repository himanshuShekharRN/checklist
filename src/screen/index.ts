import {Checklist} from './checklist';
import {Home} from './home';
import {AddList} from './addList';
import {ViewList} from './viewList';

export {Checklist, Home, AddList, ViewList};
