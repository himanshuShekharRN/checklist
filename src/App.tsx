import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
