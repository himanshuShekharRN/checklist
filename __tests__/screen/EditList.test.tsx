import React from 'react';
import {render, screen} from '@testing-library/react-native';

import {EditList} from '../../src/screen';
import {NavigationContainer} from '@react-navigation/native';
import {store} from '../../src/store';
import {Provider} from 'react-redux';

describe('Edit List Screen', () => {
  const tree = render(
    <NavigationContainer>
      <Provider store={store}>
        <EditList />
      </Provider>
    </NavigationContainer>,
  );

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('snapshot created successfully', () => {
    render(
      <NavigationContainer>
        <Provider store={store}>
          <EditList />
        </Provider>
      </NavigationContainer>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('Cancel button is present in the ViewList screen', () => {
    render(
      <NavigationContainer>
        <Provider store={store}>
          <EditList />
        </Provider>
      </NavigationContainer>,
    );
    const btnComponent = screen.getByTestId('cancelBtn');

    expect(btnComponent).toBeTruthy();
  });
});
