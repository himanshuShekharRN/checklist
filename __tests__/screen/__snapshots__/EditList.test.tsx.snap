// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit List Screen snapshot created successfully 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#ECEEF0",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#ECEEF0",
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          undefined,
        ]
      }
      testID="headersWithButton"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "opacity": 1,
              "width": 40,
            },
            {
              "backgroundColor": "transparent",
              "marginLeft": -16,
            },
          ]
        }
      >
        <View
          testID="customIcon"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000000",
                  "fontSize": 40,
                },
                undefined,
                {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="cancelBtn"
        >
          <Text
            style={
              {
                "color": "#000000",
                "fontFamily": "Roboto-Regular",
                "fontSize": 14,
              }
            }
          >
            Cancel
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#0093BB",
                "borderRadius": 30,
                "height": 45,
                "justifyContent": "center",
                "width": "33%",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": undefined,
          "height": 18,
          "width": undefined,
        }
      }
    />
    <Text
      style={
        {
          "color": "#000000",
          "fontFamily": "Roboto-Bold",
          "fontSize": 18,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": undefined,
          "height": 24,
          "width": undefined,
        }
      }
    />
  </View>
  <RCTScrollView
    bounces={false}
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#CCCDCF",
        "borderBottomWidth": 1,
        "borderTopColor": "#CCCDCF",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "paddingVertical": 22,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#005AA5",
          "borderRadius": 15,
          "height": 30,
          "justifyContent": "center",
          "marginRight": 16,
          "width": 30,
        }
      }
      testID="customIcon"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontSize": 12,
            },
            undefined,
            {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <TextInput
      autoFocus={true}
      maxLength={80}
      onChangeText={[Function]}
      style={
        {
          "color": "#000000",
          "flex": 1,
          "fontFamily": "Roboto-Regular",
          "fontSize": 16,
        }
      }
      testID="listCardTextInput"
      value=""
    />
  </View>
</RCTSafeAreaView>
`;
